### PR TITLE
.Net Passing Kernel instance to FunctionResult

### PR DESCRIPTION
**Pull request by SergeyMenshykh**
### Motivation and Context
Currently, the 'FunctionResult' class requires access to the culture configured on a kernel instance to properly perform complex type to string conversion. Since we are already passing the Kernel everywhere - including kernel functions and connectors - passing an instance of the kernel to the 'FunctionResult' class seems like a natural thing to do.

### CODE REVIEW

**Review by RogerBarreto — APPROVED** [review]

File: dotnet/samples/KernelSyntaxExamples/Example09_FunctionTypes.cs [hunk 1]
```diff
@@ -129,14 +129,14 @@ public async Task Type09Async()
     public FunctionResult Type10()
     {
         Console.WriteLine("Running function type 10");
-        return new FunctionResult(string.Empty);
+        return new FunctionResult(new Kernel(), string.Empty);
     }
 
     [KernelFunction]
     public async Task<FunctionResult> Type11Async()
```
[the remark this answers is not in the document]
Comment (SergeyMenshykh) [reply dotnet/samples/KernelSyntaxExamples/Example09_FunctionTypes.cs:136]: Yep, the kernel instance should be the same. Today, SK support Kernel as a parameter so it's easy to pass it to the result.

**Comment by SergeyMenshykh** [comment]
@RogerBarreto I've changed it to the draft to collect more feedback on the change. @stephentoub @markwallace-microsoft do you think we should proceed with this change?

**Comment by stephentoub** [comment]
> @RogerBarreto I've changed it to the draft to collect more feedback on the change. @stephentoub @markwallace-microsoft do you think we should proceed with this change?

I don't feel strongly about it. It just seems like a) if we're passing the kernel "everywhere" and it's part of a function's evaluation and some of its information is necessary in the FunctionResult then it makes sense to pass it into the FunctionResult, b) from a public API perspective it's a bit weird to expect a CultureInfo into it, and b) it's more future-proof if FunctionResult ever needs something further from the Kernel.  The downside would be if we ever need to create a FunctionResult and don't have a Kernel but still need the information it contained, and in terms of future-proofing, we can always add more constructors (including one for Kernel in the future if that's desired). So, I'm fine either way.

Does FunctionResult actually need CultureInfo, or would an IFormatProvider (which CultureInfo implements) be sufficient?

**Comment by SergeyMenshykh** [comment]
> > @RogerBarreto I've changed it to the draft to collect more feedback on the change. @stephentoub @markwallace-microsoft do you think we should proceed with this change?
> 
> I don't feel strongly about it. It just seems like a) if we're passing the kernel "everywhere" and it's part of a function's evaluation and some of its information is necessary in the FunctionResult then it makes sense to pass it into the FunctionResult, b) from a public API perspective it's a bit weird to expect a CultureInfo into it, and b) it's more future-proof if FunctionResult ever needs something further from the Kernel. The downside would be if we ever need to create a FunctionResult and don't have a Kernel but still need the information it contained, and in terms of future-proofing, we can always add more constructors (including one for Kernel in the future if that's desired). So, I'm fine either way.
> 
> Does FunctionResult actually need CultureInfo, or would an IFormatProvider (which CultureInfo implements) be sufficient?

No, the 'TypeConverter.ConvertToString(ITypeDescriptorContext context, CultureInfo culture, object value)' method needs CultureInfo and there's no overload that would accept 'IFormatProvider' instead.

I'm 50/50 as well, so I'll keep things as they are and close this draft later, unless @markwallace-microsoft has a different opinion.

**Comment by SergeyMenshykh** [comment]
We agreed, offline, to keep things as they are today and align default culture in FunctionResult and Kernel classes to be InvariantCulture to have deterministic behavior of type conversion functionality.

**Pull request closed without merging**